annotations/builder/src/main/java/io/sundr/builder/internal/utils/BuilderUtils.java

```diff
@@ -356,7 +356,10 @@ public static TypeDef getInlineReturnType(BuilderContext context, Inline inline,
       try {
         result.add(context.getElements().getTypeElement(ref.value().getCanonicalName()));
       } catch (MirroredTypeException e) {
-        result.add(context.getElements().getTypeElement(e.getTypeMirror().toString()));
+        Element element = context.getTypes().asElement(e.getTypeMirror());
```

Comment:
Shouldn't we extract the body of this method so that it's shared between both methods?

Reply:
Yeah, I think that at some point we should.